pkgs/stdenv/generic/check-meta.nix

```
@@ -133,6 +133,10 @@ let
       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allow${allow_attr} = true; }
       to ~/.config/nixpkgs/config.nix.
+
+      d) If you are using the unstable `nix` commands (e.g., `nix run`, `nix
+      build`, `nix shell`, etc.), in addition to one of the fixes above, you
+      may need to use the `--impure` flag as well.
```

Personally, I would add a little "why" this is necessary. Also word it in a way to hint that user local overlays and other settings will also be "unleashed" when using `--impure`
```suggestion
      d) If you are using the unstable `nix` commands (e.g., `nix run`, `nix
      build`, `nix shell`, etc.), then nix will evaluate in "restrict-eval" mode which
      denies it the ability to read a user's local configuration. Add `--impure`
      to enable reading from the user's nix config directory.
```

---

Thanks for the suggestion, @jonringer!